daffodil-core/src/main/scala/org/apache/daffodil/dsom/ComplexTypes.scala

```diff
@@ -84,29 +87,16 @@ sealed abstract class ComplexTypeBase(xmlArg: Node, parentArg: SchemaComponent)
   }
 }
 
-final class GlobalComplexTypeDefFactory(xmlArg: Node, schemaDocumentArg: SchemaDocument)
-  extends SchemaComponentFactory(xmlArg, schemaDocumentArg)
-  with GlobalNonElementComponentMixin {
-
-  def forElement(elementDecl: ElementDeclMixin) = new GlobalComplexTypeDef(xml, schemaDocument, elementDecl, this)
-
-  override lazy val namedQName = QName.createGlobal(name, targetNamespace, xml.scope)
-
-}
-
 /**
  * For unit testing purposes, the element argument might be supplied as null.
  */
 final class GlobalComplexTypeDef(
   xmlArg: Node,
-  schemaDocumentArg: SchemaDocument,
-  val elementDecl: ElementDeclMixin,
-  override val factory: GlobalComplexTypeDefFactory)
+  schemaDocumentArg: SchemaDocument)
   extends ComplexTypeBase(xmlArg, schemaDocumentArg)
   with GlobalNonElementComponentMixin
-  with NestingTraversesToReferenceMixin {
-
-  override lazy val referringComponent = Option(elementDecl)
+  with NestingLexicalMixin {
+  // nothing here
```

Comment:
will do. It's because the base classes/traits are providing all the necessary functionality. There is a local mixin in one class, and a global mixin in the other, but otherwise these two classes are pretty much the same, and the base class provides everything.